primitives/src/vid/advz.rs

```diff
@@ -976,6 +981,30 @@ mod tests {
         }
     }
 
+    #[test]
+    fn disperse_and_recover_with_multiplicity() {
```

Comment:
Thanks for this test. It looks like it's an "integration" test. ie. It talks only to public-facing APIs, needs no access to internal details. I suggest we remove this test and incorporate it into the existing integration test `primitives/tests/advz.rs`. You'll need to tweak the `vid_factory` closure to accept a a new `multiplicity` arg. We should add test cases for multiple distinct multiplicities, maybe all of them 1,2,4,8,16.

Reply:
Yes, I see your point but running the current tests combined (cartesian-product style) with multiplicity=1,2,4,8,16 takes 6-7min (local machine). 

I opted for a Frankenstein approach that just picks a random multiplicity for each `(payload_chunk_size, num_storage_nodes)`-pair, .

Let me know if you have something else in mind.

Reply:
good call. thank you